### ISSUE TITLE
CSV export includes comments from other Moodle components

### ISSUE THREAD

**Issue by KBorm** [drachels/moodle-mod_hotquestion#71]
One of our teachers reports that she sees comments in the hotquestion csv export that do not refer to her own course.

The reason is probably that reading comments from the Moodle comments table is only filtered by itemid. The comment itemid is not unique across Moodle. The commentarea probably needs to be included in the filter (and when writing hotquestion comments) as well.

locallib.php:
```
  // 20220818 If there are any, get the comments for each question to in the export file.
  if ($cmts = $DB->get_records('comments', ['itemid' => $q->question], 'userid, content, timecreated')) {
           .....

  }
```
This is a privacy issue. We have disabled the export function until a bugfix is available.

**Comment by drachels** [comment]
I will take a look at this as soon as I can. Right at the moment, I am trying to finalize a new Diary release, so it will probably be sometime next week, if not sooner.

**Comment by drachels** [comment]
The fix for this will be in the next release. Meanwhile, if you want to make a temporary fix, make a small addition to the line of code you list in your last entry, for the locallib.php file:
``
                // 20220818 If there are any, get the comments for each question to add in the export file.
                // 20221124 Modified to include the component.
                if ($cmts = $DB->get_records('comments', ['itemid' => $q->question, 'component' => 'mod_hotquestion'], 'userid, content, timecreated')) {
           .....

  }
``

**Comment by drachels** [comment]
Fix is in v4.1.3 and v4.1.4.
